classes/Kohana/Log/File.php

```diff
@@ -48,35 +48,20 @@ public function __construct($directory)
 	 */
 	public function write(array $messages)
 	{
-		// Set the yearly directory name
-		$directory = $this->_directory.\date('Y');
-
-		if ( ! \is_dir($directory))
-		{
-			// Create the yearly directory
-			\mkdir($directory, 02777);
-
-			// Set permissions (must be manually set to fix umask issues)
-			\chmod($directory, 02777);
-		}
-
-		// Add the month to the directory
-		$directory .= DIRECTORY_SEPARATOR.\date('m');
-
-		if ( ! \is_dir($directory))
-		{
-			// Create the monthly directory
-			\mkdir($directory, 02777);
-
-			// Set permissions (must be manually set to fix umask issues)
-			\chmod($directory, 02777);
-		}
+		// Set the year/month directory name
+		$directory = $this->_directory.\date('Y').DIRECTORY_SEPARATOR.\date('m');
```

Comment:
I can't see any reason why we should create the year and month separately? Not that we use this anyway...

Reply:
Agreed

Reply:
Ah is is that the permissions setting needs to be done on each dir in the tree individually as otherwise it would only apply to the last dir in the path?  

Additionally I don't think php's native mkdir will create parents if needed as per `mkdir -p` yet symfony/filesystem's implementation of mkdir does

Reply:
I guess if you were running under an odd umask then the permissions settings might need recursively changed. But out of the box (definitely in our env) `mkdir($path, 0775, TRUE)` will set create the parents all with the expected permissions.